### ISSUE TITLE
Vim: `n`, `gn`, `*` in workspace search multibuffer

### ISSUE THREAD

**Issue by baldwindavid** [zed-industries/zed#13437]
### Check for existing issues

- [X] Completed

### Describe the feature

I really like multibuffers resulting from a workspace search, but continually stumble over the inability to navigate and operate on them like the results of a regular buffer search. I'd like to be able to use `n`, `N`, `gn`, `*`, and `#` like in a regular buffer because that is the way I make multiple edits in a single buffer.

It's not that I've never use the built in find/replace, but probably 90% of the time my preference would be to `cgn` a result and then `n` and `.` to get eyes on and make edits throughout a multibuffer using the same vim commands as in a regular buffer. At one point `n` did navigate through the results, but think that went away with #10226 

I understand that a workspace search and a buffer search are different, but the resulting buffer and multibuffer don't really seem all that different conceptually so perhaps this could be considered.

Note that I did experiment with `n` set to `search::SelectNextMatch`. Strangely this only works in a multibuffer after first pressing `gn`.

**Comment by ceuk** [comment]
In addition to this, it would be useful to support `:s` across all files in a multi-buffer for the same reason (e.g. substitute with confirmation `%s/foo/bar/c` would be v useful)

Also, a quick mention for `:g` support in multi-buffers too if it's ever implemented as that would be highly valuable too

**Comment by everdrone** [comment]
No idea why `n` and `N` don't work in multibuffers, I find myself constantly pressing them when trying to jump around search matches

**Comment by baldwindavid** [comment]
Yes, operating on multibuffer search results is pretty well crippled in vim mode so I usually go to each excerpt in the results to make changes. It's not any easy thing to fix I would imagine though from a UI/UX perspective. We kind of need both the global search component (already there) AND a regular buffer search component in the same multibuffer to allow for the typical vim editing experience of said multibuffer.

**Comment by baldwindavid** [comment]
@ConradIrwin I read the [2025 vim roadmap post](https://zed.dev/blog/vim-2025). All the new features/plans look great, but can't help but feel like vim mode has always been broken in workspace search multibuffers. The vim mode is so nice at this point that I'd like to use it everywhere. Just mentioning this issue in case it was nowhere on the radar (perhaps the lack of votes mean it shouldn't be 😸 ). Regardless, thanks so much for your continued work on vim mode and Zed.

**Comment by ConradIrwin** [comment]
:wave: Yes! It's on my radar, see also "search improvements": https://zed.dev/channel/vim-393/notes#Non-editor-features

**Comment by baldwindavid** [comment]
Oh, I missed the link to the channel notes! Rock on.
